UefiPayloadPkg/Library/FdtParserLib/FdtParserLib.c

```diff
@@ -191,7 +197,7 @@ ParseMemory (
     if (AsciiStrCmp (TempStr, "reg") == 0) {
       Data64        = (UINT64 *)(PropertyPtr->Data);
       StartAddress  = Fdt64ToCpu (*Data64);
-      NumberOfBytes = Fdt64ToCpu (*(Data64 + 1));
+      NumberOfBytes = Fdt32ToCpu (*(Data64 + 1));

```

Comment:
there is memory node length larger than 4GB case so we cannot directly switch to Fdt32ToCpu. Maybe we could FdtSizeCells() to a global variable for rest of the code reference?

Reply:
That's what I was thinking of in the commit message. If there's a function for this, all the better

Reply:
I've moved this out to a separate branch, let's get the bulk of the work with all the general enhancements in already